forms-flow-data-analysis-api/README.md

```diff
@@ -1,64 +1,154 @@
+
+**NOTE : Default realm is `forms-flow-ai`**
+
+### Running the Application
+
+* forms-flow-data-analysis-api service uses port 5001, make sure the port is available.
+* `cd {Your Directory}/forms-flow-ai/forms-flow-data-analysis-api`
+
+* For Linux,
+  * Run `docker-compose -f docker-compose-linux.yml up -d` to start.
+* For Windows,
+  * Run `docker-compose -f docker-compose-windows.yml up -d` to start.
+
+*NOTE: Use --build command with the start command to reflect any future **.env** changes eg : `docker-compose -f docker-compose-windows.yml up --build -d`*
+
+#### To Stop the Application
+
+* For Linux,
+  * Run `docker-compose -f docker-compose-linux.yml stop` to stop.
+* For Windows,
+  * Run `docker-compose -f docker-compose-windows.yml stop` to stop.
+
+### Verify the Application Status
+
+   The application should be up and available for use at port defaulted to 5001 in <http://localhost:5001/>
+  
+* Access the **/checkpoint** endpoint for a Health Check on API to see it's up and running.
 
 ```
-     (
-        "The service at Victoria centre was completed very quickly. Also the staff was very helpful"
-        {
-            "entities": [
-                (4, 11, "SER", "pos"),
-                (15, 23, "LOC", "pos"),
-                (68, 73, "STAFF", "pos"),
-            ]
-        },
-    )
+GET http://localhost:5001/checkpoint
+
+RESPONSE
+
+{
+    "message": "Welcome to formsflow.ai Data Analysis API"
+}
 ```
 
-4. Once the data was prepared, a variety of techniques for identifying topics and sentiment analysis was tried
-out like Latent Semantic Analysis, LSTMs, BERTs. Finally, the ML model was built using a two-stage pipeline process
-to identify entities and their associated sentiment. We use a named entity recognition model(NER) to
-identify the topics, and further sentiment analysis is being done for individual entities.
+* Get the access token
 
+```
+POST {Keycloak URL}/auth/realms/<realm>/protocol/openid-connect/token
 
-5. The output of our model for the model using our ML model for a sample input is as shown below:
+Body:
+grant_type: client_credentials
+client_secret: {set client token}
+client_id: forms-flow-bpm
 
-> text="awesome location and great staff. Staff provided excellent service."
+Headers:
+Content-Type : application/x-www-form-urlencoded
 
 ```
-{
-  'sentiment': {'location': 'positive', 'facility': 'positive'},
-  'overall_sentiment': 'positive'
+
+* Access the **/sentiment** endpoint and verify response. Ensure Bearer token is passed along
+
+```
+POST http://localhost:5001/sentiment
+
+Headers:
+Content-Type : application/json
+Authorization: Bearer {access token}
+
+Payload:
+{   
+    "applicationId":{Valid application id},
+    "formUrl":"{Valid form URL}",
+    "data":[{
+        "text":"{Sample text}",
+        "elementId":""
+    }]
 }
+
 ```
 
+## Model description [xaqren/sentiment_analysis](https://huggingface.co/xaqren/sentiment_analysis)
+
```

Comment:
@seethalelias-aot here we can go with the generalized model that you trained, we don't need any service bc specific data in opensource, change that in env too

Reply:
ok